sda-commons-server-opentelemetry/src/main/java/org/sdase/commons/server/opentelemetry/OpenTelemetryBundle.java

```diff
@@ -0,0 +1,187 @@
+
+    /**
+     * Enables the bundle to setup an auto configured instance of OpenTelemetry Sdk, and register it
+     * as Global.
+     *
+     * @return the same builder.
+     */
+    FinalBuilder withAutoConfiguredTelemetryInstance();
+  }
+
+  public interface FinalBuilder {
+    /**
+     * Specifies a pattern where all traces for the matching urls will be suppressed. By default all
+     * requests with any url is traced.
+     *
+     * @param pattern The url pattern to be excluded.
+     * @return the same builder
+     */
+    FinalBuilder withExcludedUrlsPattern(Pattern pattern);
+
+    OpenTelemetryBundle build();
+  }
+
+  public static class Builder implements FinalBuilder, InitialBuilder {
+
+    private Supplier<OpenTelemetry> openTelemetryProvider;
+    private Pattern excludedUrlPatterns;
+
+    private Builder() {}
+
+    @Override
+    public FinalBuilder withTelemetryInstance(OpenTelemetry openTelemetry) {
+      this.openTelemetryProvider = () -> openTelemetry;
+      return this;
+    }
+
+    @Override
+    public FinalBuilder withExcludedUrlsPattern(Pattern excludedUrlPatterns) {
+      this.excludedUrlPatterns = excludedUrlPatterns;
+      return this;
+    }
+
+    @Override
+    public FinalBuilder withAutoConfiguredTelemetryInstance() {
+      if (shouldSkipTracing()) {
+        this.openTelemetryProvider = OpenTelemetry::noop;
+        return this;
+      }
+      this.openTelemetryProvider = OpenTelemetryBundle::bootstrapConfiguredTelemetrySdk;
+      return this;
+    }
+
+    @Override
+    public OpenTelemetryBundle build() {
+      return new OpenTelemetryBundle(openTelemetryProvider.get(), excludedUrlPatterns);
+    }
+
+    private static boolean shouldSkipTracing() {
+      // Skip loading the agent if tracing is disabled.
+      String jaegerSamplerType = getProperty(JAEGER_SAMPLER_TYPE);
+      String jaegerSamplerParam = getProperty(JAEGER_SAMPLER_PARAM);
+      boolean isDisabledUsingLegacyVars =
+          "const".equals(jaegerSamplerType) && "0".equals(jaegerSamplerParam);
+      if (isDisabledUsingLegacyVars) {
+        LOG.warn("Tracing is disabled using deprecated configuration.");
+        return true;
+      }
+      return false;
+    }
+  }
+
+  /**
+   * Decides if the {@link OpenTelemetry} instance created by {@link AutoConfiguredOpenTelemetrySdk}
+   * should be registered as the global instance. It can be troublesome for consumers of the module
+   * and will break many tests, as they have to explicitly unregister this for tests using {@link
+   * GlobalOpenTelemetry#resetForTest()} in their tests.
+   *
+   * @return the decision
+   */
+  private static boolean shouldSetAsGlobal() {
+    // Skip loading the agent if not triggered from the main thread.
+    boolean isMainThreadCheckDisabled = "false".equals(getProperty(MAIN_THREAD_CHECK_ENABLED));
+    boolean isMainThread = "main".equals(Thread.currentThread().getName());
```

Comment:
This check feels a little strange. Was it there before in the JaegerBundle?

Reply:
fixed